pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java

```diff
@@ -604,14 +606,18 @@ private static void runProducer(int producerId,
             AtomicLong numMessageSend = new AtomicLong(0);
             Semaphore numMsgPerTxnLimit = new Semaphore(arguments.numMessagesPerTransaction);
             while (true) {
+                if (produceEnough) {
```

Comment:
Can `produceEnough` be true here?

Reply:
If we produce enough messages or produce for enough time, `produceEnough` will be set to `true`.

Reply:
OK, I missed the inner-for loop